### ISSUE TITLE
fastparquet -> Hive interop trouble

### ISSUE THREAD

**Issue by brendancol** [dask/fastparquet#239]
Scenario:

I used `dask.dataframe.to_parquet` to write a parquet file.  I then expose this parquet file to Hive as an external table, but Hive is not able read int / float columns.

Example queries from Hive:
`select mystringfield from myparquettable limit 10;` --> **successfully** retrieve records
`select myIntfield from myparquettable limit 10;` --> claims myIntfield does not exist
`select myFloatfield from myparquettable limit 10;` --> claims myFloatfield does not exist

So basically all object fields work just fine for queries, but hive is unable to locate the numeric fields.  
My thought is that it is a mismatch type issue.  `float64` fields are type `DOUBLE` in hive, `int64` fields are type `BIGINT`.

When I look at `_common_metadata`, I see the metadata values are `null` while the numpy/pandas types look normal.

Have folks used fastparquet with hive?  Any tips on how to troubleshoot?

Thanks

**Comment by martindurant** [comment]
"does not exist" sounds odd - not some encoding issue, in that case.
I would first look at the dataset with fastparquet or dask itself (give the directory) to ensure that the fields do indeed exist.

**Comment by brendancol** [comment]
@martindurant hey yes they are definitely in there.  Hive just not seeing them.

**Comment by martindurant** [comment]
Did you find any work-around for this, or any clue of how hive was missing the columns?

**Comment by brendancol** [comment]
@martindurant I haven't closed this yet because I'm created a boiled down example to reproduce the issue.

In the short-term, I ended up using `pyarrow` to write the parq file and that fixed the issue.  I'm going to be digging in a bit more though and will add the reproduction case to this issue.  Thx a million for your help, turned out `pyspark` was able to read in the parquet file and the re-write an version which was readable by hive.

**Comment by cpbhagtani** [comment]
@brendancol did you find out the root cause of interop issue with hive? We are hitting the same issue with boolean types as well.

**Comment by winglian** [comment]
I am seeing the same issue where a simple `SELECT id FROM my_tbl` errors with `does not contain requested field: optional int64 id`. But to throw another wrench in it, presto can read the same file without any issues.

**Comment by martindurant** [comment]
So is `optional int64 id` not the right type for that field?

**Comment by cobbr2** [comment]
I'm seeing the same behavior.  Schema:

**Comment by antonellyb** [comment]
> @martindurant I haven't closed this yet because I'm creating a boiled down example to reproduce the issue.
> 
> In the short-term, I ended up using `pyarrow` to write the parq file and that fixed the issue. I'm going to be digging in a bit more though and will add the reproduction case to this issue. Thx a million for your help, turned out `pyspark` was able to read in the parquet file and the re-write an version which was readable by hive.



> @martindurant I haven't closed this yet because I'm creating a boiled down example to reproduce the issue.
> 
> In the short-term, I ended up using `pyarrow` to write the parq file and that fixed the issue. I'm going to be digging in a bit more though and will add the reproduction case to this issue. Thx a million for your help, turned out `pyspark` was able to read in the parquet file and the re-write an version which was readable by hive.

@brendancol 
Could you please explain how you created parquet file with pyarrow?
I tried the following way, but hive cannot read it as well 
```
import pyarrow.parquet as pq
import pyarrow as pa
table = pa.Table.from_pandas(df)
pq.write_table(table, `'example.parquet')
````

**Comment by yohplala** [comment]
Quickly reading the ticket, it seems the request still open here is this one:

> The parquet files were dumped originally in August 2018; I find it disturbing that the created-by string 1.0.0 (build 111) has been [hardcoded in the source](https://github.com/dask/fastparquet/blob/8767c2f94975aed8d052a9d0ad6f5e4e630a6f12/fastparquet/util.py#L20 for three years rather than derived from anything git-ish, and so is completely useless for determining which specific version of `fastparquet` did the deed.

**Comment by martindurant** [comment]
Sure, the string could in theory reflect the library version. However, it's not sued anywhere, so only for the eyes of users.
